### PR TITLE
feat(emotion): Upgrade emotion to v10

### PR DESCRIPTION
**Pull request by billyvg**
Read the [migration log](https://emotion.sh/docs/migrating-to-emotion-10) for more information.

From their migration guide...

> The most significant change in Emotion 10 is that it doesn’t let you easily access the underlying class names. Instead of thinking in class names, you have to think in terms of styles and composing them together (you can still use class names directly if you want to, but you won’t get new features like zero-config server rendering).
> For example, in Emotion 9, you might have had two classes from css and compose them together with cx, but in Emotion 10, you create two styles and compose them together in the css prop like this:

Take a look at [this diff](https://github.com/getsentry/sentry/pull/15106/commits/c4650bd0df147f75fa78d9f2f15743a3450f3380#diff-a356f9e124b894ee6c144dd054abe349R145) for an example.

### grid-emotion
We are still deprecating this, but in the meantime we are upgrading to `reflexbox` so that we can push forward with the emotion upgrade without having to refactor all of grids out.

### Build times
~No measurements yet~ Roughly 9.8% increase in build time. We may be able to mitigate impact on incremental builds a bit by forking the type checking (although this will not help with your energy bill): https://github.com/getsentry/sentry/pull/15587

This is the `tsc --diagnostics` comparison

### master
```
Files:           995
Lines:        466265
Nodes:        820647
Identifiers:  274490
Symbols:      618451
Types:        207662
Memory used: 703338K
I/O read:      0.21s
I/O write:     0.25s
Parse time:    2.41s
Bind time:     0.73s
Check time:   13.54s
Emit time:     2.44s
Total time:   19.12s
```

### Emotion 10
```
Files:          1002
Lines:        466702
Nodes:        822659
Identifiers:  275206
Symbols:      697918
Types:        582946
Memory used: 994339K
I/O read:      0.20s
I/O write:     0.26s
Parse time:    2.56s
Bind time:     0.90s
Check time:   30.12s
Emit time:     2.43s
Total time:   36.01s
```

### Bundle sizes
Roughly 1.8% increase in filesize (gzipped).

## TODO
- [x] Update `grid-emotion` eslint rules (https://github.com/getsentry/eslint-config-sentry/pull/47)
- [x] Update `getsentry`

### CODE REVIEW

**Comment by billyvg** [comment]
This PR may be easier to review by looking at commit ranges. The last 2 commits are the ones that contain bulk changes via codemods (e.g. new `styled` imports and new `grid-emotion` changes).

**Comment by billyvg** [comment]
I will do a staging deploy to do some clicking around this week and will ping slack for some help.

**Comment by EvanPurkhiser** [comment]
Are the reflexbox changes a codemod too?

**Comment by billyvg** [comment]
They were a combination of my own codemod and search and replace for the library name itself.

**Comment by ckj** [comment]
Very cool. Any particular reason for using reflexbox over rebass/grid?

**Comment by billyvg** [comment]
@ckj @rebass/grid is now reflexbox: https://github.com/rebassjs/grid

**Comment by ckj** [comment]
> @ckj @rebass/grid is now reflexbox: https://github.com/rebassjs/grid

🤯

**Review by dashed — APPROVED** [review]

File: src/sentry/static/sentry/app/components/tooltip.tsx [hunk 1]
```diff
@@ -136,22 +136,24 @@ class Tooltip extends React.Component<Props, State> {
...
     // it with a span tag so that popper has ref
-    if (children.type instanceof Function) {
-      propList.containerDisplayMode = this.props.containerDisplayMode;
-      return (
-        <Container {...propList} innerRef={ref}>
-          {children}
-        </Container>
-      );
+
+    if (typeof children.type === 'string') {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/tooltip.tsx]: @markstory Does this seem right?

File: src/sentry/static/sentry/app/utils/theme.tsx [hunk 2]
```diff
@@ -302,6 +302,8 @@ const theme = {
     empty: '#e2dee6',
     colors: ['#ec5e44', '#f38259', '#f9a66d', '#98b480', '#57be8c'],
   },
+
+  space: [0, 8, 16, 20, 30],
```
Comment (billyvg) [inline src/sentry/static/sentry/app/utils/theme.tsx]: This is so that `reflexbox` uses the same spacing as our `space`

**Review by ckj — APPROVED** [review]

File: webpack.config.js [hunk 3]
```diff
@@ -322,6 +322,13 @@ let appConfig = {
   resolve: {
     alias: {
       app: path.join(staticPrefix, 'app'),
+      '@emotion/styled': path.join(staticPrefix, 'app', 'styled'),
```
Comment (billyvg) [inline webpack.config.js]: Oh thoughts on this @EvanPurkhiser? We need to do this to get typing for `styled`

Comment (EvanPurkhiser) [reply webpack.config.js]: Ah I see, seems fine.

Comment (EvanPurkhiser) [reply webpack.config.js]: Maybe drop a comment in here for why we do this

**Pull request merged**